### PR TITLE
chore(updatecli): Adds documentation to the script and enhances the `jq` command.

### PR DESCRIPTION
**Pull request by gounthar**
This is a follow-up PR to #907.

I refined the tag selection logic in `jq` by implementing a filter for tags containing hyphens.
To enhance code understanding, I added explanatory comments that provide additional context about the data retrieved from the RedHat REST API.

A comprehensive markdown documentation detailing the API's functionality is currently in development and will be included in an upcoming pull request.

### Testing done
```
./updatecli/scripts/ubi9-latest-tag.sh
9.5-1732804088
```
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

**Review by dduportal — DISMISSED** [review]
Checks says:

```
✔ Bump UBI9 version:
	Source:
		✔ [latestVersion] Get the latest UBI9 Linux version
	Condition:
		✔ [checkUbi9DockerImage] Check if the container image "ubi9" is available
	Target:
		✔ [updateDockerBake] Update the default value of the variable UBI9_TAG in the docker-bake.hcl
		✔ [updateDockerfile] Update the value of the base image (ARG UBI9_TAG) in the Dockerfile
```


That looks good to me!

Let's have a secondary review by @lemeurherve as he asked for some of these changes in previous PRs

**Review by lemeurherveCB — COMMENTED** [review]
Thanks for the follow-up!

Some suggestions below.

File: updatecli/scripts/ubi9-latest-tag.sh [hunk 1]
```diff
@@ -16,8 +28,13 @@ if ! command -v jq >/dev/null 2>&1 || ! command -v curl >/dev/null 2>&1; then
...
 fi
 
-# Fetch the tags using curl
-response=$(curl --silent --fail --location --connect-timeout 10 --max-time 30 --header 'accept: application/json' "$URL")
+# Fetch the tags using curl, for `ub9`, in the `registry.access.redhat.com`, sorted by last update date descending, and keeping only page 0.
+# --fail: Fail silently on HTTP errors
+# --silent: Don't show progress meter or error messages
+# --show-error: Show error message if it fails
+# --retry 3: Retry failed requests up to 3 times
+# --retry-delay 2: Wait 2 seconds between retries
```
[the remark this answers is not in the document]
Comment (lemeurherveCB) [reply updatecli/scripts/ubi9-latest-tag.sh]: > A comprehensive markdown documentation detailing the API's functionality is currently in development and will be included in an upcoming pull request.

For RedHat API? If so, I don't think that's the place either as it's not a core part of Jenkins agents, I would suggest not spending time on this.

**Comment by lemeurherveCB** [comment]
Reposting a comment from an automatically hidden thread:

> A comprehensive markdown documentation detailing the API's functionality is currently in development and will be included in an upcoming pull request.

For RedHat API? If so, I don't think it has a place in this repo as it's not a core part of Jenkins agents, I would suggest not spending time on this.

**Review by lemeurherveCB — APPROVED** [review]
LGTM, thx!

File: updatecli/scripts/ubi9-latest-tag.sh [hunk 2]
```diff
@@ -16,8 +27,8 @@ if ! command -v jq >/dev/null 2>&1 || ! command -v curl >/dev/null 2>&1; then
     exit 1
 fi
 
-# Fetch the tags using curl
-response=$(curl --silent --fail --location --connect-timeout 10 --max-time 30 --header 'accept: application/json' "$URL")
+# Fetch the tags using curl, for `ubi9`, in the `registry.access.redhat.com`, sorted by last update date descending, and keeping only the first page.
```
Comment (lemeurherveCB) [inline updatecli/scripts/ubi9-latest-tag.sh:30]:
```suggestion
# Fetch `ubi9` from registry.access.redhat.com sorted by most recent update date, and keeping only the first page.
```
suggestion(nit/optional): shorter formulation, and keep the link clickable

**Review by MarkEWaite — APPROVED** [review]

**Review by dduportal — APPROVED** [review]

**Pull request merged**